src/EventSubscriber/UserUpdatedSubscriber.php

```diff
@@ -34,7 +34,10 @@ public function onFlush(OnFlushEventArgs $args): void
 
         foreach ($unitOfWork->getScheduledEntityUpdates() as $entity) {
             $changes = $unitOfWork->getEntityChangeSet($entity);
-            if ($entity instanceof User && \array_key_exists('email', $changes)) {
+
+            if ($entity instanceof User &&
+            (\array_key_exists('email', $changes) // if email has changed
+            || (\array_key_exists('roles', $changes) && \in_array('ROLE_USAGER', $changes['roles'][0])))) { // is usager becomes user
```

Comment:
`if` plutôt que `is` dans le commentaire ?

Reply:
done